platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/reconnect/VirtualTeacherTreeView.java

```diff
@@ -54,11 +54,52 @@ public final class VirtualTeacherTreeView<K extends VirtualKey, V extends Virtua
 
     private static final Logger logger = LogManager.getLogger(VirtualTeacherTreeView.class);
 
+    // A queue of the nodes (by path) that we are about to handle. Note that ConcurrentBitSetQueue
+    // cleans up after itself in "chunks", such that we don't end up consuming a ton of memory.
+    // We use two separate accumulating and processing queues to implement a level-by-level traversal:
+
+    /** A queue to which we add items for the next level currently. */
+    private ConcurrentBitSetQueue accumulatingHandleQueue = new ConcurrentBitSetQueue();
+
+    /** A queue which we're processing currently (aka sending nodes to the learner). */
+    private ConcurrentBitSetQueue processingHandleQueue = new ConcurrentBitSetQueue();
+
+    /** Flip the queues by exchanging the accumulating and processing queues. */
+    private synchronized void flipQueues() {
+        final ConcurrentBitSetQueue temp = accumulatingHandleQueue;
```

Comment:
An assertion like this would be helpful here:
```java
    assert processingHandleQueue.isEmpty();
```

Reply:
Added.